prompterator/utils.py

```diff
@@ -311,3 +333,8 @@ def insert_hidden_html_marker(helper_element_id, target_streamlit_element=None):
         """,
         unsafe_allow_html=True,
     )
+
+
+def format_multiline_text_for_markdown(text):
+    text = re.sub(r" ", "&nbsp;", text)
```

Comment:
I'm not particularly proud of this 🤦 but I use it preserve the exact formatting of the text. The thing is, `st.error` treats the text as markdown, which means collapsing consecutive spaces into one, which makes stack traces completely unreadable (all the indentation is lost). With this approach long lines do get wrapped in our `st.error` (but you made me check this, so thank you!). In the worst case, it even splits a word at an arbitrary point just to wrap the line.

Reply:
Note that we could actually use "```" blocks in the markdown to preserve all formatting of stack traces but that's exactly the approach that leads to long lines overflowing without getting wrapped, so it really messes up the entire Prompterator UI in case of basically any stack trace (long lines guaranteed 🤦).

Reply:
I see, thanks for the explanation. Let's leave it then as is maybe with one tiny change - could we call the method somewhat differently, e.g. `format_stacktrace_for_markdown` so that we know to not use it for regular texts? 😊